capsules/src/date_time.rs

```diff
@@ -0,0 +1,350 @@
+//! Real Time Clock driver
+//!
+//! Allows handling of the current date and time
+//!
+//! Author: Irina Bradu <irinabradu.a@gmail.com>
+//!
+//!
+//! Usage
+//! -----
+//!
+//! ```rust
+//!  let grant_dt = create_capability!(capabilities::MemoryAllocationCapability);
+//!  let grant_date_time = board_kernel.create_grant(capsules::date_time::DRIVER_NUM, &grant_dt);
+//!
+//!  let date_time = static_init!(
+//!     capsules::date_time::DateTime<'static>,
+//!     capsules::date_time::DateTime::new(&peripherals.rtc, grant_date_time)
+//!  );
+//!  kernel::hil::date_time::DateTime::set_client(&peripherals.rtc, date_time);
+//! ```
+
+use crate::driver::NUM;
+use core::cell::Cell;
+use kernel::grant::{AllowRoCount, AllowRwCount, Grant, UpcallCount};
+use kernel::hil::date_time;
+
+use kernel::errorcode::into_statuscode;
+use kernel::syscall::{CommandReturn, SyscallDriver};
+use kernel::utilities::registers::{register_bitfields, LocalRegisterCopy};
+use kernel::{ErrorCode, ProcessId};
+
+pub const DRIVER_NUM: usize = NUM::DateTime as usize;
+
+pub enum DateTimeCommand {
+    ReadDateTime,
+    SetDateTime,
+}
+
+#[derive(Default, Clone, Copy)]
+pub struct AppData {
+    subscribed: bool,
+}
+
+pub struct DateTime<'a> {
+    date_time: &'a dyn date_time::DateTime<'a>,
+    apps: Grant<AppData, UpcallCount<1>, AllowRoCount<0>, AllowRwCount<0>>,
+    in_progress: Cell<bool>,
+}
+
+register_bitfields![u32,
```

Comment:
It would be good to document that these aren't actually registers, it's just using register definitions for bitfields.